src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -306,5 +362,26 @@ private ImmutableArray<CodeRefactoringProvider> CreateRefactorings(string langua
                 return builder.ToImmutableAndFree();
             }
         }
+
+        private sealed class CodeRefactoringServiceCallback
+        {
+            private readonly CodeRefactoringService _codeRefactoringService;
+
+            public CodeRefactoringServiceCallback(CodeRefactoringService codeRefactoringService)
+            {
+                _codeRefactoringService = codeRefactoringService;
+            }
+
+            public TextSpan? TryGetPastedTextSpan(SourceTextContainer sourceTextContainer)
+            {
+                if (_codeRefactoringService._pasteTrackingService is null)
+                    return null;
+
+                if (_codeRefactoringService._pasteTrackingService.TryGetPastedTextSpan(sourceTextContainer, out var textSpan))
+                    return textSpan;
+
+                return null;
+            }
```

Comment:
this is a bit concerning that we're exposing a synchronous method over to the OOP side.  in general we don't do that.  can this be made async?